src/Controller/HybridAuthController.php

```diff
@@ -98,10 +101,20 @@ public function authenticate($provider_id) {
    */
   public function endpoint() {
     try {
-      $provider = new LinkedIn($this->getConfiguration());
+      $session = $this->request->getSession();
+
+      // Saving provider id in session for further usage.
+      $provider_id = $session->get('hybridauth_provider');
+      // Configure function name and path for it.
+      $provider_function = 'Hybridauth\\Provider\\' . $provider_id;
```

Comment:
I see it 2nd time...DX should be improved.

Reply:
Added condition class_exists().